File: sources/fortran/02_switch_case/main.f90
! task 02 switch_case — expected output: 7500000075000000
! build: gfortran -O3 -o prog main.f90 (flang -O3 -o prog main.f90)    run: ./prog

program main
  implicit none
  integer(kind=8) :: acc, i

  acc = 0_8
  do i = 0_8, 99999999_8
    select case (mod(i, 4_8))
    case (0_8)
      acc = acc + 1_8
    case (1_8)
      acc = acc + i
    case (2_8)
      acc = acc + 2_8*i
    case (3_8)
      acc = acc + 3_8*i
    end select
  end do

  write(*,'(i0)') acc
end program main

File: sources/ruby/02_switch_case/main.rb
# task 02 switch_case — expected output: 7500000075000000
# build: none (interpreted)    run: ruby main.rb (cruby) | ruby --yjit main.rb (cruby+yjit) | jruby main.rb (jruby, needs Java 25)
# The stock Windows CRuby build has no YJIT: `ruby --yjit` warns "Ruby was built without YJIT support".

acc = 0
i = 0
while i < 100_000_000
  case i % 4
  when 0 then acc += 1
  when 1 then acc += i
  when 2 then acc += 2 * i
  when 3 then acc += 3 * i
  end
  i += 1
end

puts acc
